File: JavascriptPOO/exercices/corrections/TP-heritage-vehicule/classes/vehicule.js
export class Vehicule {
    constructor(ma, m, k, a) {
        this.marque = ma
        this.model = m
        this.km = k
        this.annee = a
    }

    display() {
        return `Marque: ${this.marque}, Model: ${this.model}, Km: ${this.km}, annee: ${this.annee}`
    }
}

File: FormationDotNet/CoursCSharpPOO/Classes/Vehicule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CoursCSharpPOO.Classes
{
    abstract class Vehicule
    {
        private string marque;
        private string model;

        public string Marque { get => marque; set => marque = value; }
        public string Model { get => model; set => model = value; }

        public Vehicule(string marque, string model)
        {
            Marque = marque;
            Model = model;
        }

        public abstract bool Demarrer();
        public abstract void Arreter();

        public abstract void FairePlein(double volume);
    }
}
